source/win32_tinyengine.c

```diff
@@ -5,6 +5,234 @@
 
 #include "tinyengine_types.h"
 
+typedef struct win32_digital_button win32_digital_button;
+struct win32_digital_button
+{
+    u32 Up;
+    u32 Down;
+    b32 Pressed;
+    b32 Released;
+
+    b32 WMDown;
+};
+
+typedef struct win32_mouse win32_mouse;
+struct win32_mouse
+{
+    win32_digital_button Left;
+    win32_digital_button Middle;
+    win32_digital_button Right;
+    win32_digital_button Extra1;
+    win32_digital_button Extra2;
+    s16 WheelDelta;
+    s32 X; // TODO(hayden): Vector, or leave as is?
+    s32 Y; // TODO(hayden): Vector, or leave as is?
+    f32 NormX; // TODO(hayden): Is this the normalization Kyryl was requesting?
+    f32 NormY;
+};
+
+// NOTE(hayden): The order here matters for calculation from KEY_0 to KEY_DOWN
+typedef enum te_keys te_keys;
+enum te_keys
```

Comment:
Prefer typedef enum name { dashjdkasd; }  name;

Reply:
This enum could be not platform specific i imagine? Maybe introduce a tinyengine_platform.h so other people can use this enum